workflows/prognostic_c48_run/runtime/emulator.py

```diff
@@ -0,0 +1,185 @@
+import dataclasses
+import xarray as xr
+from typing import (
+    Callable,
+    Hashable,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Iterable,
+    Set,
+    Union,
+)
+
+from fv3fit.emulation.thermobasis.emulator import Config as EmulatorConfig
+from fv3fit.emulation.thermobasis.xarray import get_xarray_emulator
+from runtime.monitor import Monitor
+from runtime.names import SPHUM, DELP
+from runtime.types import State, Diagnostics, Step
+
+__all__ = ["PrognosticAdapter", "Config"]
+
+
+def strip_prefix(prefix: str, variables: Iterable[str]) -> Set[str]:
+    return {k[len(prefix) :] for k in variables if k.startswith(prefix)}
+
+
+@dataclasses.dataclass
+class Config:
+    """
+    Attributes:
+        checkpoint: Either path to model artifact in Weights and biases
+            "<entity>/<project>/<name>:tag" to be loaded, or the configurations
+            for a new emulator object.
+    """
+
+    emulator: Union[str, EmulatorConfig] = dataclasses.field(
+        default_factory=EmulatorConfig
+    )
+    # online parameters
+    online: bool = False
+    train: bool = True
+    # will ignore the emulator for any z larger than this value
+    # remember higher z is lower in the atmosphere hence "below"
+    ignore_humidity_below: Optional[int] = None
+
+
+@dataclasses.dataclass
+class PrognosticAdapter:
+    """Wrap a Step function with an emulator
+
+    The wrapped function produces diagnostic outputs prefixed with
+    ``self.emulator_prefix_`` and trains/applies the emulator to ``state``
+    depending on the user configuration.
+
+    Attributes:
```

Comment:
I didn't document everything since mostly these can be inferred from the type. I added a description of `timestep` since its units are not not clear from the typehints alone.